src/vma/iomux/epfd_info.cpp

```diff
@@ -348,8 +348,9 @@ int epfd_info::add_fd(int fd, epoll_event *event)
 			events |= EPOLLOUT;
 		}
 		if (events != 0) {
-			insert_epoll_event(fd, events); // mutex is recursive
+			insert_epoll_event(fd, events, false); // mutex is recursive
 		}
+		do_wakeup();
```

Comment:
Code should be narrowed to just this:
```code
if (events != 0) {
		insert_epoll_event(fd, events); // mutex is recursive
  }
else {
	do_wakeup();
}
```

Instead of changing other code places.